fiftyone/__public__.py

```diff
@@ -51,7 +51,7 @@
     Metadata,
     ImageMetadata,
 )
-from .core.sample import Sample
+from .core.sample import FrameSample, Sample
```

Comment:
I'm still pondering what the right class name for `FrameSample` is.....

I'm leaning towards `SampleFrame`, because it represents information about the frame of a sample.

Also, to state a bit more about why the notion that `FrameSample` derives from `Sample` bothers me is that `FrameSample` is not _really_ a `Sample` on its own. It doesn't have its default fields such as `filepath`, `tags`, `metadata`, etc, for example. Those are derived from its parent `Sample`.

Also, since `FrameSample` corresponds to a particular frame number, how about storing the frame number in the document itself? Like how `VideoFrameLabels` in ETA stores the frame number in itself, even though the dict that contains it also encodes the frame number (the latter is a string, which is potentially another reason to store a copy of the frame number in the frame doc)

This doesn't necessarily need to be addressed immediately, although at least the naming will affect the public interface. It would be nice for that not to change.

I haven't actually played with the data structures yet in code; or looked too carefully below, but, out of curiosity, how are default fields handled in `FrameSample` currently?

Reply:
Update: tried myself and see that the normal `Sample` default fields are hidden on `FrameSample`; that's good

Reply:
I've updated this to just be `Frame`. I intend on documenting first thing tomorrow, when my words will make more sense.